### PR TITLE
Blockchain application off-chain meta endpoint does not adhere to the limits

### LINKED ISSUE #1609: Blockchain application off-chain meta endpoint does not adhere to the limits
### Actual behavior
Querying the `/api/v3/blockchain/apps/meta?limit=5` returns 7 items while the limit is 5:
```
"meta": {
	"count": 7,
	"offset": 0,
	"total": 7
}
```
### Expected behavior

Querying the endpoint `/api/v3/blockchain/apps/meta?limit=5` must return exactly 5 items if limit is 5.

### Steps to reproduce

- Query `/api/v3/blockchain/apps/meta?limit=5` with limit request params.

### Which version(s) does this affect? (Environment, OS, etc...)
v0.7.0-beta.0

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #1609 

### How was it solved?

- [x] Sliced array to adhere to limit passed
- [x] Correct offset calculation incase isDefault is not passed as a parameter
- [x] Fixed integration test cases

### How was it tested?

- [x] Integration tests
- [x] APIs

### CODE REVIEW

File: services/blockchain-app-registry/shared/metadata.js [hunk 1]
```diff
@@ -181,16 +181,31 @@ const getBlockchainAppsMetadata = async (params) => {
...
 		const nonDefaultApps = await applicationMetadataTable.find(
 			{ ...params, limit, isDefault: false },
 			['network', 'appDirName', 'isDefault'],
 		);
 
+		// Reset the offset to the original value
+		params.offset = originalOffset;
+
 		blockchainAppsMetadata.data.push(...nonDefaultApps);
 	}
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadata.js:204]: I'd prefer to not modify the params.
```suggestion
	if (params.isDefault !== true && blockchainAppsMetadata.data.length < params.limit) {
		let offset = { params };

		// If params.isDefault is not passed in the request then adjust the offset
		if (!('isDefault' in params)) {
			const totalDefaultApps = await applicationMetadataTable.count(
				{ ...params, limit, isDefault: true },
			);

			offset = (params.offset - totalDefaultApps) > 0 ? params.offset - totalDefaultApps : 0;
		}

		const nonDefaultApps = await applicationMetadataTable.find(
			{ ...params, offset, limit, isDefault: false },
			['network', 'appDirName', 'isDefault'],
		);

		blockchainAppsMetadata.data.push(...nonDefaultApps);
	}
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please consider the following.

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Pull request merged**